packages/rendermime/src/rendermime.ts

```diff
@@ -80,58 +85,64 @@ class RenderMime implements IRenderMime {
   }
 
   /**
-   * Get an iterator over the ordered list of mimeTypes.
-   *
-   * #### Notes
-   * These mimeTypes are searched from beginning to end, and the first matching
-   * mimeType is used.
+   * The sanitizer used by the rendermime instance.
+   */
+  readonly sanitizer: ISanitizer;
+
+  /**
+   * The ordered list of mimeTypes.
    */
-  mimeTypes(): IIterable<string> {
-    return new ArrayIterator(this._order);
+  get mimeTypes(): ReadonlyArray<string> {
+    return this._mimeTypes;
   }
 
   /**
-   * Render a mime model.
+   * Create a renderer for a mime model.
    *
-   * @param model - the mime model to render.
+   * @param mimeType - the mime type to render.
    *
    * #### Notes
-   * Renders the model using the preferred mime type.  See
+   * Creates a renderer widget using the preferred mime type.  See
    * [[preferredMimeType]].
    */
-  render(model: IRenderMime.IMimeModel): IRenderMime.IReadyWidget {
-    let mimeType = this.preferredMimeType(model);
-    if (!mimeType) {
-      return this._handleError(model);
+  createRenderer(mimeType: string, trusted: boolean): IRenderMime.IRendererWidget {
+    let factory = this._factories[mimeType];
+    if (!factory) {
+      throw new Error(`Unregistered mimeType: ${mimeType}`);
     }
-    let rendererOptions = {
+    let options = {
       mimeType,
-      model,
+      trusted,
       resolver: this._resolver,
       sanitizer: this.sanitizer,
       linkHandler: this._handler
     };
-    return this._renderers[mimeType].render(rendererOptions);
+    if (!factory.canCreateRenderer(options)) {
+      throw new Error('Cannot create renderer');
+    }
+    return factory.createRenderer(options);
   }
 
   /**
    * Find the preferred mimeType for a model.
    *
    * @param model - the mime model of interest.
    *
+   * @param trusted - Whether the model is trusted.
+   *
    * #### Notes
    * The mimeTypes in the model are checked in preference order
-   * until a renderer returns `true` for `.canRender`.
+   * until a renderer returns `true` for `.canCreateRenderer`.
    */
-  preferredMimeType(model: IRenderMime.IMimeModel): string {
+  preferredMimeType(model: IRenderMime.IMimeModel, trusted: boolean): string {
     let sanitizer = this.sanitizer;
-    return find(this._order, mimeType => {
-      if (model.data.has(mimeType)) {
-        let options = { mimeType, model, sanitizer };
-        let renderer = this._renderers[mimeType];
+    return find(this._mimeTypes, mimeType => {
+      if (mimeType in model.data) {
+        let options = { mimeType, sanitizer, trusted };
+        let renderer = this._factories[mimeType];
         let canRender = false;
         try {
-          canRender = renderer.canRender(options);
+          canRender = renderer.canCreateRenderer(options);
```

Comment:
If it is a third party factory, we don't want it to affect our ability to render a mime bundle .  We added this because the `ipywidgets` renderer did raise an exception.